src/vscode-bicep/src/commands/insertResource.ts

```diff
@@ -16,14 +15,10 @@ export class InsertResourceCommand implements Command {
     context: IActionContext,
     documentUri?: Uri
   ): Promise<void> {
-    documentUri = await findOrCreateActiveBicepFile(
-      context,
-      documentUri,
-      "Choose which Bicep file to insert a resource into"
-    );
-
-    const document = await workspace.openTextDocument(documentUri);
-    const editor = await window.showTextDocument(document);
+    const editor = await findOrShowEditor(documentUri);
+    if (!editor) {
+      throw new Error(`Unable to find active editor for insert resource command`);
```

Comment:
A little confused on the fix here. Are we avoiding calling the `findFiles()` method by omitting the `findOrCreateActiveBicepFile()` method above, and then implementing a quicker way of finding an active editor?

Reply:
My observation was that launching the file picker dialog is **ridiculously** slow - so the main goal of this PR is to avoid using a file picker dialog at all. `findOrCreateActiveBicepFile()` has logic to launch a file picker, which is why I'm replacing it with my own implementation.

The "Insert Resource" command can be executed in two ways:
* Right-click -> "Insert Resource" (inside an active Bicep file)
    ![image](https://user-images.githubusercontent.com/38542602/210889153-b8252118-f58c-4e1c-b085-103a9c41b542.png)
* Ctrl+Shift+P -> "Insert Resource":
    ![image](https://user-images.githubusercontent.com/38542602/210889396-264c011f-5f7a-4bab-a498-23bf3b665642.png)

In the first case, the user is executing the command in the context of an active document, so `documentUri` will be set. This works fine - I'm not making any changes here.

In the second case, the user is executing the command globally in the window - so `documentUri` will be `null` on the API call we get from VSCode. It's now up to us to figure out which file the user was attempting to execute the command for. The change I'm making in this PR to package.json means that this command can **only** be executed if there is an active `.bicep` file - previously it could be executed even if there is no active Bicep file. As a result, it is now completely unambiguous - `vscode.window.activeTextEditor` will give us the active file that the user executed the command for, and there's no need to launch a dialog.

Reply:
Also, FYI I'm putting this PR on hold until we've had a chance to discuss it in the Bicep Discussions meeting (see Stephen's comment)